framework/src/main/java/org/checkerframework/framework/source/SourceChecker.java

```diff
@@ -1676,6 +1965,77 @@ protected final void setLintOption(String name, boolean val) {
     }
   }
 
+  /**
+   * Returns the set of subchecker classes on which this checker depends. Returns an empty set if
+   * this checker does not depend on any others.
+   *
+   * <p>Subclasses should override this method to specify subcheckers. If they do so, they should
+   * call the super implementation of this method and add dependencies to the returned set so that
+   * checkers required for reflection resolution are included if reflection resolution is requested.
+   *
+   * <p>If you wish to create a checker that runs multiple checkers, but itself is not a type
+   * system, then subclass {@link AggregateChecker}.
+   *
+   * <p>If a checker should be added or not based on a command line option, use {@link
+   * #getOptionsNoSubcheckers()} or {@link #hasOptionNoSubcheckers(String)} to avoid recursively
+   * calling this method.
+   *
+   * <p>Each subchecker of this checker may also depend on other checkers. If this checker and one
+   * of its subcheckers both depend on a third checker, that checker will only be instantiated once.
+   *
+   * <p>Though each checker is run on a whole compilation unit before the next checker is run, error
+   * and warning messages are collected and sorted based on the location in the source file before
+   * being printed. (See {@link #printOrStoreMessage(Diagnostic.Kind, String, Tree,
+   * CompilationUnitTree)}.)
+   *
+   * <p>WARNING: Circular dependencies are not supported nor do checkers verify that their
+   * dependencies are not circular. Make sure no circular dependencies are created when overriding
+   * this method. (In other words, if checker A depends on checker B, checker B cannot depend on
+   * checker A.)
+   *
+   * <p>This method is protected so it can be overridden, but it should only be called internally by
+   * {@link SourceChecker}.
+   *
+   * @return the subchecker classes on which this checker depends; will be modified by callees
+   */
+  // This is never looked up in, but it is iterated over (and added to, which does a lookup).
+  protected Set<Class<? extends SourceChecker>> getImmediateSubcheckerClasses() {
```

Comment:
Is the contrast between "depends" and "subchecker" intentional?  What difference is it emphasizing?

Reply:
Yes, this method is only dealing with subcheckers that this checker uses.  Any of those subchecker may also have subcheckers, but this method does not return those.

Reply:
Can we eliminate the new term "depends", so that we don't have to define it, and use "immediate subchecker" throughout instead?